strax/plugin.py

```diff
@@ -847,3 +847,51 @@ def cleanup(self, iters, wait_for):
             for s in savers:
                 s.close(wait_for=wait_for)
         super().cleanup(iters, wait_for)
+
+
+@export
+class CutPlugin(Plugin):
+    """Generate a plugin that provides a boolean for a given cut specified by 'cut_by'"""
+    save_when = SaveWhen.NEVER
+
+    def __init__(self):
+        super().__init__()
+
+        _name = strax.camel_to_snake(self.__class__.__name__)
+        if not hasattr(self, 'provides'):
+            self.provides = _name
+        if not hasattr(self, 'cut_name'):
+            self.cut_name = _name
+        if not hasattr(self, 'cut_description'):
+            _description = _name
+            if 'cut_' not in _description:
+                _description = 'Cut by ' + _description
+            else:
+                _description = " ".join(_description.split("_"))
+            self.cut_description = _description
+
+    def infer_dtype(self):
+        dtype = [(self.cut_name, np.bool_, self.cut_description)]
+        # Alternatively one could use time_dt_fields for low level plugins.
+        dtype = dtype + strax.time_fields
+        return dtype
+
+    def compute(self, **kwargs):
+        if hasattr(self, 'cut_by'):
+            cut_by = self.cut_by
+        else:
+            raise NotImplementedError(f"{self.cut_name} does not have attribute 'cut_by'")
+
+        # Take shape of the first data_type like in strax.plugin
+        buff = list(kwargs.values())[0]
+
+        # Generate result buffer
+        r = np.zeros(len(buff), self.dtype)
+        r['time'] = buff['time']
```

Comment:
You have to add a test whether buff will be an empty. In that case buff['time'] will crash.

Reply:
Actually, you can make copies of empty data. For example the line below does not fail:
```
strax.endtime(np.zeros(0, dtype=strax.time_fields))
```